File: frontend/src/models/user_endpoint_usage.ts
import { Endpoint } from "./endpoint";
import { User } from "./user";

export interface UserEndpointUsage {
    ID: number;
    UserID: number;
    EndpointID: number;
    RequestCount: number;
    LastAccess: string;
    CreatedAt: string;
    UpdatedAt: string;

    User: User;
    Endpoint: Endpoint;
}

File: backend/models/user_endpoint_usage.go
package models

import (
	"time"
)

type UserEndpointUsage struct {
	ID           uint      `gorm:"primaryKey"`
	UserID       uint      `gorm:"not null"`
	EndpointID   uint      `gorm:"not null"`
	RequestCount int       `gorm:"default:0"`
	LastAccess   time.Time `gorm:"not null"`
	CreatedAt    time.Time
	UpdatedAt    time.Time

	User     User     `gorm:"foreignKey:UserID;constraint:OnDelete:CASCADE"`
	Endpoint Endpoint `gorm:"foreignKey:EndpointID;constraint:OnDelete:CASCADE"`
}
